### ISSUE TITLE
Global search: accented search queries don't return any results

### ISSUE THREAD

**Issue by EmilePerron** [bolt/core#2427]
In the documentation, the following statement is made:
> Searching is case insensitive, and normalised. This means that a search for 'CAFE' or 'cafe' is equal, and either will show records that have the word 'Café' in it.

However, in our project, the normalized aspect doesn't seem to hold up.

Details
-------

| Question                      | Answer
|----------------- | ------
| Relevant Bolt Version | 4.1
| Install type                   | Composer install
| BC Break                      | no
| PHP version                 | 7.3
| Web server                  | Apache
| Database                     | MySQL

Reproduction
------------

### Bug summary

In the global search, searching for any word with accented characters returns no results at all. 

For example, if you have a page with a redactor field that contains the following text:
> En s’abonnant au Musée du Fjord, vous contribuerez à l’élargissement des publics du Musée, à la qualité et à l’originalité de ses productions, ainsi qu’à son rayonnement international. Être abonné du Musée donne aussi droit à de nombreux privilèges.

You would expect the page to show up when searching for the words `privilèges` or `originalité`. However, it does not.

### Specifics

I haven't tested on any other versions, but the problem occurred on Bolt 4.1.13.

At first glance, this seems to be due to the UTF-8 encoding of accented characters in the `value` field of the `bolt_field_translation`. 

The runnable database query for the search, based on Symfony's profiler, looks like this:
```sql
SELECT b0_.id AS id_0, b1_.id AS id_1, b1_.name AS name_2, b1_.sortorder AS sortorder_3, b1_.version AS version_4, b1_.type AS type_5 
FROM bolt_content b0_ 
INNER JOIN bolt_field b1_ ON b0_.id = b1_.content_id AND b1_.type IN ('generic', 'block', 'number', 'collection', 'embed', 'email', 'select', 'imagelist', 'templateselect', 'set', 'textarea', 'html', 'markdown', 'hidden', 'slug', 'file', 'filelist', 'checkbox', 'text', 'date', 'image', 'data', 'article', 'seo', 'redactor') 
INNER JOIN bolt_field_translation b2_ ON b1_.id = b2_.translatable_id 
WHERE b2_.value LIKE '%privilèges%' 
ORDER BY b1_.sortorder ASC;
```

However, the `value` of that redactor field in `bolt_field_translation` looks like this:
```json
["<p>En s\u2019abonnant au Mus\u00e9e du Fjord, vous contribuerez \u00e0 l\u2019\u00e9largissement des publics du Mus\u00e9e, \u00e0 la qualit\u00e9 et \u00e0 l\u2019originalit\u00e9 de ses productions, ainsi qu\u2019\u00e0 son rayonnement international. \u00catre abonn\u00e9 du Mus\u00e9e donne aussi droit \u00e0 de nombreux privil\u00e8ges.<\/p>"]
```

Seeing as special characters are encoded, the `LIKE` fails to return the expected results.

### Possible solutions
A few possible solutions come to mind:

- Encoding the search query used in the SQL request (ex.: `WHERE b2_.value LIKE '%privil\u00e8ges%' `)
- Adding a column that contains the indexed content in `bolt_field_translation` instead of searching in the JSON encoded `value`column
- Using the `JSON_UNESCAPED_UNICODE` flag when using `json_encode()` (seems like it could break quite a few things)

**Comment by EmilePerron** [comment]
I briefly discussed the issue with Bob on Slack: the first suggested solution (_Encoding the search query used in the SQL request_) is the simplest and least risky of the bunch, so we'll go with that. 

I'll work on this and submit a PR soon.

**Comment by bobdenotter** [comment]
Fixed in #2435
